### ISSUE TITLE
Inferred ints can take floats in a release build

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#42124]
**Godot version:**

3.2.3 rc6
Not sure about master

**Issue description:**

Check this code:
```
extends Node2D

var test1 := 120
var test2: int = 120

func _ready() -> void:
	test1 = 120.15
	test2 = 120.15
	$Label.text = str(test1)
	$Label2.text = str(test2)
```
When you run it in editor or debug build, both labels show 120. In a release build, the first label shows 120.15

**Minimal reproduction project:**

[ReproductionProject.zip](https://github.com/godotengine/godot/files/5234367/ReproductionProject.zip)

**Comment by Calinou** [comment]
I can reproduce this on 3.2.2 (official Linux release export template).

**Comment by KoBeWi** [comment]
Impossible to test in master due to #42127

**Comment by JestemStefan** [comment]
Can't reproduce in https://downloads.tuxfamily.org/godotengine/testing/4.0/4.0-dev.20210727/

I mean it shows 120 in release build.
![obraz](https://user-images.githubusercontent.com/37214990/128931791-e8fdd873-d52a-4e61-bb38-b48db2197de0.png)

Here is a project with exports to check. I hope it helps

**Comment by JestemStefan** [comment]
In 3.3.2:
![obraz](https://user-images.githubusercontent.com/37214990/128933203-51bfed7d-b83d-4acb-9ff0-396b5f9160df.png)

`var test1 := 120` fail
`var test2: int = 120` works

**Comment by akien-mga** [comment]
Confirmed fixed in `master`, so this is a 3.x only issue now.

**Comment by akien-mga** [comment]
Fixed by #57851.
